pvlib/iotools/tmy.py

```diff
@@ -171,21 +169,10 @@ def read_tmy3(filename=None, coerce_year=None, recolumn=True):
 
     head = ['USAF', 'Name', 'State', 'TZ', 'latitude', 'longitude', 'altitude']
 
-    if str(filename).startswith('http'):
-        request = Request(filename, headers={'User-Agent': (
-            'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) '
-            'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 '
-            'Safari/537.36')})
-        response = urlopen(request)
-        csvdata = io.StringIO(response.read().decode(errors='ignore'))
-    else:
-        # assume it's accessible via the file system
-        csvdata = open(str(filename), 'r')
+    csvdata = open(str(filename), 'r')
```

Comment:
removing the `if` statement gives us an opportunity to easily improve the code using a `with` statement. I think the following would work:

```python
with open(filename, 'r') as csvdata:
    firstline = csvdata.readline()
    meta = dict(zip(head, firstline.rstrip('\n').split(",")))
    data = pd.read_csv(csvdata, header=0)
```